pkg/asset/manifests/gcp/cluster.go

```diff
@@ -235,3 +235,20 @@ func getSubnet(ctx context.Context, project, region, subnetName string) (*comput
 
 	return subnet, nil
 }
+
+// GetTagsFromInstallConfig will return a slice of ResourceManagerTags from UserTags in install-config.
+func GetTagsFromInstallConfig(installConfig *installconfig.InstallConfig) []capg.ResourceManagerTag {
+	var tags []capg.ResourceManagerTag
+	if len(installConfig.Config.Platform.GCP.UserTags) > 0 {
+		tags = make([]capg.ResourceManagerTag, len(installConfig.Config.Platform.GCP.UserTags))
+		for i, tag := range installConfig.Config.Platform.GCP.UserTags {
+			tags[i] = capg.ResourceManagerTag{
```

Comment:
Are we doing any kind of validation here to ensure that the tags are meeting the capg (and probably gcp) requirements? I didn't see any in the installconfig/gcp/userTags.go file.

Reply:
The userTags feature has the tags validation here - https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/gcp/usertags.go#L116
But you're right, its missing validations on the Tag fields, for example the key limits:
```
maxLength: 63
minLength: 1
pattern: ^[a-zA-Z0-9]([0-9A-Za-z_.-]{0,61}[a-zA-Z0-9])?$
```

I'm not sure we want to add those checks in this PRs, seems like it should be done in a separate PR, with separate bug, as its not related to CAPG specifically.

Reply:
Note also that there is kubebuilder validations on the ResourceManagerTag so invalid values will be rejected by CAPG https://github.com/kubernetes-sigs/cluster-api-provider-gcp/blob/main/api/v1beta1/tags.go#L34-L37